core/store/src/trie/state_snapshot.rs

```diff
@@ -177,7 +178,7 @@ impl ShardTries {
     pub fn create_state_snapshot(
         &self,
         prev_block_hash: CryptoHash,
-        shard_uids: &[ShardUId],
+        shard_uids: &[(ShardIndex, ShardUId)],
```

Comment:
nit: `shard_uids` doesn't fit 100% anymore as the parameter name
some mostly bad options: `shards_indexes_and_uids`, `shards_info`, `shards_details`, ..